src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -52,25 +59,65 @@ export class MarkdownToolbarComponent {
 	_toggleMarkdownViewAction: ToggleViewAction;
 
 	constructor(
+		@Inject(INotebookService) private _notebookService: INotebookService,
 		@Inject(IInstantiationService) private _instantiationService: IInstantiationService,
-		@Inject(IContextMenuService) private contextMenuService: IContextMenuService
-	) { }
+		@Inject(IContextMenuService) private contextMenuService: IContextMenuService,
+		@Inject(IConfigurationService) private _configurationService: IConfigurationService
+	) {
+		super();
+		this._register(this._configurationService.onDidChangeConfiguration(e => {
+			this.previewFeaturesEnabled = this._configurationService.getValue('workbench.enablePreviewFeatures');
```

Comment:
You'll need to update the action bar too when this happens otherwise only new toolbars will get it